Makefile

```diff
@@ -0,0 +1,64 @@
+LOCAL_IMAGE = acorn_docker:1.0
+REMOTE_IMAGE = merlinran/acorn_docker
+
+.PHONY: list
+list:
+	@echo "Welcome! You may want to try one of the following:\n---"; \
+	grep "^.PHONY:" Makefile | grep "#" | cut -d ":" -f 2- | sed "s/\w*#/\t#/g" | sed "s/^/make/"
+
+.PHONY: simulation # Run the vehicle and server containers in simulation mode.
+simulation: docker-image
+	@DOCKER_COMPOSE_FILE=docker-compose-simulation.yml make restart-docker-compose &&\
+	echo "Please visit http://localhost"
+
+.PHONY: attach-vehicle # Attach to the shell of the vehicle container. It creates the container if it doesn't exist.
+attach-vehicle:
+	@test -z `docker ps -f "name=acorn_vehicle" -q` && make docker-vehicle; \
```

Comment:
Here's again plain shell. Start the container if it doesn't exist, then `docker exec` it. `test -z` returns `0`(success) if the output of the command is empty, in which case `make docker-vehicle` will run; Otherwise it returns `1`(fail), and the `&&` short-circuits. The command after `;` is not affected, hence being run unconditionally.